### ISSUE TITLE
time.Time.Format does not provide consistency for space padding

### ISSUE THREAD

**Issue by hallazzang** [golang/go#22802]
Please answer these questions before submitting your issue. Thanks!

### What version of Go are you using (`go version`)?
`go version go1.9.2 darwin/amd64`

### Does this issue reproduce with the latest release?
I'm not sure I'm running the latest release of Go, but I just ran `brew upgrade go` on my macOS laptop and it says `go 1.9.2 already installed`.

### What operating system and processor architecture are you using (`go env`)?
```
GOARCH="amd64"
GOOS="darwin"
```

### What did you do?
I was trying to make a library that gives programmers who wanted to use Python-like `strftime` method on `time.Time` a nice `Strftime` function. While writing test code to test if paddings are working, I found that there is no space(literal ` `) padding for 'month' and 'hour' but only 'day' haves it.

To explain what happens, here's Go playground's link: https://play.golang.org/p/cAgWgVD5Tu

As you can see underscore(`_`) layout string doesn't work for month, hour, minute and second.

### What did you expect to see?
In the above playground, I expected ` 9  4` for month and day.

### What did you see instead?
But I've got `_9  4` instead. Same for month, hour, minute and second.

I would like to patch the `time` package and I'd like to know whether this issue has already issued or not. Thanks.

**Comment by bradfitz** [comment]
Well, this is documented at least:

> Within the format string, an underscore _ represents a space that may be replaced by a digit if the following number (a day) has two digits; for compatibility with fixed-width Unix time formats.

Up to @robpike to decide whether to extend it to support other things.

**Comment by hallazzang** [comment]
Yes, and I've found another thing that 24-hour based hour cannot represented without padding since `15`(already has two digits) is proper placeholder for it. Would it be solved, too?

**Comment by robpike** [comment]
It does seem inconsistent but is it a common enough case to worry about? No one has noticed the issue before and it's easy to fix by post-processing.

What do others think?

**Comment by hallazzang** [comment]
Well I was surprised that no one has notice this issue but me, I think the reason is that the zero padding fits well for almost every usecases. But in my opinion a consistency for standard packages matters a lot, would it good to patch it by myself and send a pull request to this repository? I've never contributed to this large repository before, though.

**Comment by robpike** [comment]
This repository does not accept pull requests directly. Please see https://golang.org/doc/contribute.html

**Comment by hallazzang** [comment]
Thank you. I'll read it and get on my work soon!

**Comment by hallazzang** [comment]
By the way, how do you think about the 24-based hour's no-padding(`7` instead of ` 7` for minute) representation? Currently it seems impossible to correctly support consistent padding rule for 24-based hour since it already has 2 digits(`15`). How could one workaround this?

**Comment by rsc** [comment]
Rob will review.

**Comment by hallazzang** [comment]
@rsc Anything should I do for now?
